MPCAutofill/cardpicker/integrations/patreon.py

```diff
@@ -81,28 +88,76 @@ def get_patrons(campaign_id: str, campaign_tiers: dict[str, SupporterTier]) -> O
         return None
 
     try:
-        members = requests.get(
-            # https://docs.patreon.com/#get-api-oauth2-v2-campaigns-campaign_id-members
-            url=f"https://www.patreon.com/api/oauth2/v2/campaigns/{campaign_id}/members",
-            params={
-                "include": "currently_entitled_tiers",
-                "fields[member]": ",".join(
-                    ["full_name", "campaign_lifetime_support_cents", "pledge_relationship_start", "patron_status"]
-                ),
-            },
-            headers=patreon_header,
-        ).json()["data"]
+        req = (
+            requests.get(
+                # https://docs.patreon.com/#get-api-oauth2-v2-campaigns-campaign_id-members
+                f"https://www.patreon.com/api/oauth2/v2/campaigns/{campaign_id}/members",
+                params={
+                    "include": "currently_entitled_tiers",
+                    "fields[member]": ",".join(
+                        ["full_name", "campaign_lifetime_support_cents", "pledge_relationship_start", "patron_status"]
+                    ),
+                },
+                headers=patreon_header,
+            )
+            if next_page is None
+            else requests.get(next_page, headers=patreon_header)
+        )
+        res = req.json()
+
+        # Ready the next page if provided
+        next_page = res.get("links", {}).get("next")
 
         # Return formatted list of patrons
-        return [
-            {
-                "name": mem["attributes"]["full_name"],
-                "tier": campaign_tiers[mem["relationships"]["currently_entitled_tiers"]["data"][0]["id"]]["title"],
-                "date": mem["attributes"]["pledge_relationship_start"][:10],
-            }
-            for mem in members
-            if len(mem["relationships"]["currently_entitled_tiers"]["data"]) > 0
-        ]
+        results: list[Supporter] = []
+        for mem in res["data"]:
+
+            # Skip non-active members
+            if mem["attributes"]["patron_status"] != "active_patron":
+                continue
+
+            # Skip members with no tiers
+            tiers = mem.get("relationships", {}).get("currently_entitled_tiers", {}).get("data", [])
+            if not tiers:
+                continue
+
+            # Figure out highest paid tier
```

Comment:
it's slightly clearer if you do
```suggestion
            # Figure out highest paid tier for this patron (patrons can subscribe to multiple tiers)
```